ui2/src/common/TableTransfer.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import { Transfer } from 'antd';
+import _ from 'lodash';
+import { AppTable } from './Table';
+import { Link } from 'react-router-dom';
+import { AppButton } from '../common/Button';
+import { tableColumns } from '../utils/tableColumns';
+import './TableTransfer.css';
+
+export const TableTransfer = ({
+  data,
+  searchColumns,
+  rightColTitles,
+  leftColTitles,
+  selectGroup,
+  predefinedGroups,
+  linkTo,
+  addSearch,
+  ...props
+}) => {
+  const [targetKeys, setTargetKeys] = useState([]);
+  const [showSearch] = useState(!!searchColumns);
+  const formattedData = data.map(dataItem => ({ ...dataItem, key: dataItem.id.toString() }));
+
+  const handleSearch = (inputValue, item) => {
+    return searchColumns.map(searchCol => item[searchCol] && item[searchCol].indexOf(inputValue) !== -1).includes(true);
+  };
+  const handleGroupSelection = predefinedGroupId => {
+    const keys = selectGroup(predefinedGroupId);
+    const addedKeys = _.difference(keys, targetKeys);
+    if (addedKeys.length) {
+      setTargetKeys([...targetKeys, ...addedKeys]);
+    } else {
+      setTargetKeys(_.difference(targetKeys, keys));
+    }
+  };
+
+  return (
+    <Transfer
+      className="table-transfer"
+      dataSource={formattedData}
+      filterOption={(inputValue, item) => handleSearch(inputValue, item)}
```

Comment:
can't you just pass 
 filterOption=handleSearch
?

Reply:
Ofcourse, my bad... fixed!